clients/fides-js/src/lib/preferences.ts

```diff
@@ -49,6 +50,10 @@ export const updateConsentPreferences = ({
     });
   });
 
+  // Update the cookie object
+  // eslint-disable-next-line no-param-reassign
+  cookie.consent = consentCookieKey;
```

Comment:
This mutates the provided `cookie` param, which might be a *bit* surprising... but I think it makes sense to do this as we don't store that state anywhere else super declaratively. This is especially true since we call `saveFidesCookie` below which persists the updated state to the browser - it'd be surprising to me to find that the `cookie` object was actually more stale than the version saved to browser since we _weren't_ updating that object before!

Reply:
this is fine. It's the same pattern we use in the `init` method of `fides.ts`